profile-widget/profilewidget2.cpp

```diff
@@ -1610,8 +1610,9 @@ void ProfileWidget2::addDivemodeSwith()
 	QAction *action = qobject_cast<QAction *>(sender());
 	QPointF scenePos = mapToScene(mapFromGlobal(action->data().toPoint()));
 	for (i = 0; i < UNDEF_COMP_TYPE; i++)
-		if (QString(divemode_text[i]) == action->text())
-			add_event(current_dc, lrint(timeAxis->valueAt(scenePos)), 8, 0, i, "modechange");
+		if (QString(tr(divemode_text[i])) == action->text())
+			add_event(current_dc, lrint(timeAxis->valueAt(scenePos)), 8, 0, i,
+				QT_TRANSLATE_NOOP("gettextFromC", "modechange"));
```

Comment:
Don't translate event names.

Reply:
I guess there is some misunderstanding regarding QT_TRANSLATE_NOOP. This doesn't really translate the string but only pushes it to the translation system. For the position where it is called it is a NO-OP.
It makes it possible to translate the string later-on for the UI only. For the event names this is done here:
https://github.com/sfuchs79/subsurface/blob/master/profile-widget/diveeventitem.cpp#L149

That is a strategy you could maybe also use in your PR #1289 and which would maybe help to avoid having to duplicate the const char *divemode_text...

Reply:
No, forget the last sentence I wrote. At the moment we want to change from "OC" to "Open circuit" for the UI we maybe really need two versions of the string list. Otherwise we may change the XML from "OC" to "Open cuircuit" as well and this would lead to the kind of incompatibility @willemferguson mentioned.

Reply:
I still don't understand why and where we'd want "modechange" to be translated. Can you explain why you are marking it for translation?

Reply:
@dirkhh 
My understanding is that we translate it like other event names when we display it on the UI in the info box in the profile. Relevant code for the translation is here:
https://github.com/sfuchs79/subsurface/blob/master/profile-widget/diveeventitem.cpp#L149

I tested this right now and this part works:
![grafik](https://user-images.githubusercontent.com/25316050/41505371-35505f34-7208-11e8-8ba5-2eaa7d744a25.png)

BUT the dive mode names (here "open circuit") are still nowhere translated and I don't know why...

Reply:
My guess is because I haven't pulled the translations :)

Reply:
OK, translations pulled, try now

Reply:
No, that is not the root cause. I downloaded from Transifex and merged the German language file manually for my test.

But I fixed it and unfortunately another PR will follow. I once again was lost with all those different tr() "domains". We already once had this discussion before. If I want to access a string provided in a C file from a CPP file I need to use
`gettextFromC::instance()->tr();` instead of a simple `tr();`